### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by Ghilain**
### Project requirements:

- Create a class `Classroom`
- Create the relationship between `Classroom `and `Student`
- Create a class `Book`
- Create a class `Rental`
- Create the relationship between `Person `and `Book `using the intermediate class `Rental`

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -27,4 +27,8 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(date, person)
+    Rental.new(date, person, self)
+  end
```
Comment (Deepakdanger) [inline person.rb]: - [x] In this `add_rental`, I think you need to pass `book` and not `person`. Also, on the line 32, the first argument will be `book`, then person (i.e. self), and last `date`.

File: person.rb [hunk 2]
```diff
@@ -27,4 +27,8 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(date, person)
+    Rental.new(date, person, self)
```
Comment (Deepakdanger) [inline person.rb]: - [x] In the `person.rb` you also need `rentals` as an instance variable just like in the `book.rb` to make the association. Also, getters and setters should be present for `rentals`.

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Rental
+  attr_accessor :date
+
```
Comment (Deepakdanger) [inline rental.rb]: - [x] Getters and Setters for `@book` and `@person` instance variables should be present.

File: classroom.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  attr_accessor :label, :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students.push(student)
+  end
```
Comment (Deepakdanger) [inline classroom.rb]: - [x] Also setup the has-many association with student in the add_student method(a classroom has many students)).

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Ghilain, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 4 above)
Comment (Ghilain) [reply classroom.rb]: Thanks @Deepakdanger

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @Ghilain , 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 





Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ghilain** [comment]
Thanks @zieeco

**Pull request merged**